File: Proxy01.js
// Objeto Real

class ServicoReal {
    executar() {
      console.log("Executando o serviço real...");
    }
  }
  
  // Proxy
  
  class ProxyServico {
    constructor(){
    this.servicoReal = new ServicoReal();

    }
  
    executar() {
      console.log("Antes da execução do serviço.");
      this.servicoReal.executar();
      console.log("Depois da execução do serviço.");
    }
  }
  
  // Uso do proxy cliente
  
  const proxy = new ProxyServico();
  proxy.executar();
  

File: Proxy01.py
# Objeto Real

class ServicoReal:
    def executar(self):
        print("Executando o serviço real...")

# Proxy

class ProxyServico:
    def __init__(self):
        self.servico_real = ServicoReal()

    def executar(self):
        print("Antes da execução do serviço.")
        self.servico_real.executar()
        print("Depois da execução do serviço.")

# Uso do proxy cliente

proxy = ProxyServico()
proxy.executar()
